Spdssii 18 Server-Side

cschroed-usgs opened the pull request:
Adding new spatial membership and network attributes to the serialization.

cschroed-usgs commented on sparrow-business/src/test/java/gov/usgswim/sparrow/service/DomainSerializerTest.java:
```diff
@@ -0,0 +1,183 @@
+		
+		
+	}
+	
```
Just a helper method for subsequent tests. This will make more sense if you look at the usage below first.

cschroed-usgs commented on sparrow-business/src/test/java/gov/usgswim/sparrow/service/DomainSerializerTest.java:
```diff
@@ -0,0 +1,183 @@
+		String result = writer.toString();
+		return result;
+	}
+	
```
The following tests are not rigorous -- they merely check that strings contain marker text. Two better ways would be to 1) use xpath assertions, or 2) build a model parser. I think both of those are too time-consuming given schedule constraints. Are there any low-hanging fruits that I'm not considering for testing an xml serializer more rigorously?

smlarson-usgs replied:
There's a test class you may want to check out that does parsing: ParserHelperTest found in the business project.

cschroed-usgs commented on sparrow-business/src/test/java/gov/usgswim/sparrow/service/DomainSerializerTest.java:
```diff
@@ -0,0 +1,183 @@
+
+	@Before
+	public void setUp() {
+		mb = new SparrowModelBuilder();
```
Most values are not important or significant to tests -- they are merely plausible.

cschroed-usgs commented on sparrow-business/src/main/java/gov/usgswim/sparrow/service/DomainSerializer.java:
```diff
@@ -91,9 +91,32 @@ private void readModel() {
 				addNonNullBasicTag("constituent", model.getConstituent());
 				addNonNullBasicTag("units", model.getUnits().getUserName());
+				addNonNullBasicTag("baseYear", Integer.toString(model.getBaseYear()));
+				addOpenTag("spatialMembership");
```
Omitting the states and regions if it is a national model is ok because the current UI paradigm is that a user's region/state selections will show {the relevant region and the national models} or the {relevant state and the national models}.

cschroed-usgs commented:
Renamed a class called "SparrowModelParser" to "SparrowModelRequestParser" to more accurately reflect the purpose of that class and to distinguish it from the new class that actually parses serialized sparrow models.

cschroed-usgs commented on sparrow-business/src/main/java/gov/usgswim/sparrow/service/DomainSerializer.java:
```diff
@@ -1,197 +0,0 @@
-package gov.usgswim.sparrow.service;
```
Renamed this. Not sure why it is showing up as a deletion.

smlarson-usgs replied:
Looks like a great start. Would you please add a comment on what remains to be done so when its picked up in the future, it will be easier to implement?

cschroed-usgs replied:
Good thinking! Pseudocode added!

cschroed-usgs commented:
To maintain a single source of truth for multiple classes, I extracted strings like tag and attribute names from ModelSerializer to constants in a class called ModelSerializerConstants. I updated ModelSerializer to use the constants. I wrote a partial ModelParser implementation that uses those constants.
